Userland/Libraries/LibELF/DynamicLinker.cpp

```diff
@@ -450,17 +450,24 @@ static Result<void*, DlErrorMessage> __dlsym(void* handle, const char* symbol_na
     __pthread_mutex_lock(&s_loader_lock);
     ScopeGuard unlock_guard = [] { __pthread_mutex_unlock(&s_loader_lock); };
 
-    auto object = static_cast<DynamicObject*>(handle);
-    if (!handle) {
-        auto library_name = get_library_name(s_main_program_name);
-        auto global_object = s_global_objects.get(library_name);
-        object = *global_object;
-    }
-    auto symbol = object->lookup_symbol(symbol_name);
-    if (!symbol.has_value()) {
-        return DlErrorMessage { String::formatted("Symbol {} not found", symbol_name) };
+    if (handle) {
+        auto object = static_cast<DynamicObject*>(handle);
+        auto symbol = object->lookup_symbol(symbol_name);
+        if (symbol.has_value()) {
+            return symbol.value().address.as_ptr();
+        }
+    } else {
+        // When handle is 0 (RTLD_DEFAULT) we should look up the symbol in all global modules
+        // https://pubs.opengroup.org/onlinepubs/009604499/functions/dlsym.html
+        for (auto& entry : s_global_objects) {
```

Comment:
Quick question here; is this hash map an ordered hash map? And is it inserted into in the correct global module resolution order? If either of those properties are missing this is going to give very very strange symbol resolution results for the applications that depend on it.

Reply:
The hash map type itself is ordered, as for the insertion order... Well, upon closer inspection I found that `DynamicLinker` already has static `lookup_global_symbol()` method which iterates over this hash map in the same manner and does what I wanted to do here, but more correctly. 

Updated the branch to use this method instead of reinventing it.